### PR TITLE
feat(ui5-color-palette): set selected color to inner color picker

### LINKED ISSUE #8772: [ui5-color-palette-popover]: ColorPicker does not take the last selected color from palette
### Bug Description

Hi, team.

If I select a color in the palette (from default colors or from recent colors) and then I choose to open `More Colors...`, the ColorPicker has the last used color of the Picker, not from the Palette. If no color has been selected in the Picker, then white is predefined.

### Affected Component

_No response_

### Expected Behaviour

Selecting colors in the palette would reflect in the picker when choosing "More Colors..."

### Isolated Example

_No response_

### Steps to Reproduce

1. Add `ui5-color-palette-popover` with `showRecentColors`, `showMoreColors` and `showDefaultColor`;
2. Define some `IColorPaletteItem` in the default slot;
3. Select a color from the palette;
4. Choose `More Colors...` to open the ColorPicker;
5. Check that white is predefined, or the last picked color is used, but not the last palette color selected.

### Log Output, Stack Trace or Screenshots

_No response_

### Priority

Medium

### UI5 Web Components Version

1.20

### Browser

Chrome, Edge, Safari

### Operating System

_No response_

### Additional Context

_No response_

### Organization

SAP Event Ticketing

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

### PR DESCRIPTION
**Pull request by NHristov-sap**
Until now, when a color is being selected in `ui5-color-palette` or `ui5-color-palette-popover`, and the inner `ui5-color-picker` is being opened via **More Colors...** button, the selected color was not shown in the picker nor in the color area or color sliders, nor as RGBA values.

This PR adds this feature and now when there is selected color in the color palette/color palette popover, and the color picker is activated, it displays the selected color and its RGBA values.

Fixes #8772

### CODE REVIEW

File: packages/main/cypress/specs/ColorPalette.cy.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+			.find("#alpha")
+			.as("alpha");
+
+		cy.get<ColorPalette>("@colorPalette")
+			.shadow()
+			.find("ui5-dialog")
+			.find("ui5-button")
+			.as("okButton");
+
+		cy.get<ColorPaletteItem>("@redColor")
```
Comment (didip1000) [inline packages/main/cypress/specs/ColorPalette.cy.ts:65]: maybe take out the action to open and check for the color values in a command, or just the check

Comment (didip1000) [reply packages/main/cypress/specs/ColorPalette.cy.ts:65]: Also you can try splitting up the tests into smaller ones, if possible.

**Review by didip1000 — CHANGES_REQUESTED** [review]

File: packages/main/cypress/specs/ColorPalette.cy.ts (hunk 1 above)
Comment (NHristov-sap) [reply packages/main/cypress/specs/ColorPalette.cy.ts]: I have created a command to test a single color.

**Review by didip1000 — APPROVED** [review]

File: packages/base/src/util/ColorConversion.ts [hunk 2]
```diff
@@ -196,6 +196,19 @@ const getRGBColor = (color: string): ColorRGB => {
 	return HEXToRGB(color);
 };
 
+const getAlpha = (color: string): number => {
+	let alpha = 1;
+
+	if (color.startsWith("rgba")) {
```
Comment (tsanislavgatev) [inline packages/base/src/util/ColorConversion.ts:202]: we should either change it to getRGBAAlpha or we should also have hsla added to this check.

File: packages/base/src/util/ColorConversion.ts [hunk 3]
```diff
@@ -196,6 +196,19 @@ const getRGBColor = (color: string): ColorRGB => {
...
+		const parts = color.split(",");
```
Comment (tsanislavgatev) [inline packages/base/src/util/ColorConversion.ts:203]: Also for a solution, we could use reggex if we only want rgba:

const rgbaRegex = /^rgba\(\s*\d+\s*,\s*\d+\s*,\s*\d+\s*,\s*([\d.]+)\s*\)$/;
const match = rgbaString.match(rgbaRegex);

    if (match) {
        return parseFloat(match[1]);
    }

But that's if we don't have hsla and rgba

File: packages/main/src/ColorPalette.ts [hunk 4]
```diff
@@ -502,6 +502,12 @@ class ColorPalette extends UI5Element {
 
 	_openMoreColorsDialog() {
 		const dialog = this._getDialog();
+		const colorPicker = this.getColorPicker();
+		const value = this._currentlySelected ? this._currentlySelected.value : undefined;
+
+		if (value) {
+			colorPicker.value = value;
+		}
```
Comment (tsanislavgatev) [inline packages/main/src/ColorPalette.ts:510]: isn't this the same as 

colorPicker.value = this._currentlySelected ? this._currentlySelected.value : undefined;

**Review by tsanislavgatev — CHANGES_REQUESTED** [review]

File: packages/base/src/util/ColorConversion.ts (hunk 2 above)
Comment (NHristov-sap) [reply packages/base/src/util/ColorConversion.ts:202]: I have added also check for hsla.

File: packages/base/src/util/ColorConversion.ts (hunk 3 above)
Comment (NHristov-sap) [reply packages/base/src/util/ColorConversion.ts:203]: I am willing to keep the non-regex method as it is more readable.

File: packages/main/src/ColorPalette.ts (hunk 4 above)
Comment (NHristov-sap) [reply packages/main/src/ColorPalette.ts:510]: Nope, as the colorPicker **value** property cannot accept `undefined` as it is of type **string**  but **_currentlySelected** `value` property is optional.

**Pull request merged**